site/docs/tech/graph.md

```diff
@@ -0,0 +1,100 @@
+Then if there is a parent with depth lower than transaction one by `MaxDepthDiff` round, such parent is considered invalid and error is returned.
+If transaction's depth is not bigger exactly by one than max depth of parents - then such parents considered invalid. 
+In the opposite case - transaction considered as the one which has valid parents.
+
+#### Parents selection
+As was mentioned in the begining, potential parents transaction are stored in separate index - balanced tree in particular. Transaction are sorted by depth and in case of equal depth - by seed length ([seed](#transactionseed)).
+Tree is traversed in descending order, i.e. higher depth goes first. First `MaxParentsPerTransaction` number of eligible parents will be selected as parents for new transaction. If parent has depth lower than graph by `MaxDepthDiff` rounds or has children which present in the graph and "complete" such parent won't be taken and instead will be removed from parents index in the graph.
+In other words we prioritise "leaf" transactions to be parents.  
+
+##### Algorithm
+1. Descend eligible parents balanced tree
+    * if there is transaction found which has lower depth than graph height by `MaxDepthDiff` round it should not be selected as parent and deleted from parents index
+    * if there is transaction found which has children and at least one of them is neither missing nor incomplete, such transaction should not be selected as parent and should be deleted from parents index
+2. Add transaction to selected parents
+3. If there are exactly `MaxParentsPerTransaction` parents found then return those, otherwise continue to #1 
+
+#### Critical transaction selection
+Potential [critical transactions](#transactioncriticality) stored in separate index in form of balanced tree, same way as eligible parent transactions - sorted by depth first and seed length second.
+Tree is traversed in ascending order, i.e. lower depth and higher seed len first. First critical transaction with depth higher than root's will be taken. Meanwhile if there will be found transactions during traversal which have depth lower or equal than root's or which aren't critical for given difficulty, they will be removed from this index.
+
+##### Algorithm
+1. Ascend eligible critical balanced tree
+    * if transaction depth is lower or equal than root depth - such transaction cannot be selected as critical and should be deleted from critical index
+    * if transaction is not critical for given difficulty - such transaction cannot be selected as critical and should be deleted from critical index
+2. Return found critical transaction or nil if its not found
+
+#### Transaction Seed
+Transaction [seed](https://github.com/perlin-network/wavelet/blob/master/tx.go#L55) its a value which is computed as [blake2b](https://en.wikipedia.org/wiki/BLAKE_(hash_function)#BLAKE2) 256 hash checksum from transaction sender id and all parent transaction ids.
```

Comment:
Care to place the equation for the seed condition in LaTeX?

Reply:
i'm not familiar with latex, but if its needed i'll check how to do it.

Reply:
Hmm okay, it's not a problem then :+1: